plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, List, NamedTuple, Optional, Union
+
+import duckdb
+import pandas as pd
+import pyarrow as pa
+
+from flytekit import PythonInstanceTask
+from flytekit.extend import Interface
+from flytekit.types.structured.structured_dataset import StructuredDataset
+
+
+class QueryOutput(NamedTuple):
+    counter: int = -1
+    output: Optional[str] = None
+
+
+class DuckDBQuery(PythonInstanceTask):
+    _TASK_TYPE = "duckdb"
+
+    def __init__(
+        self,
+        name: str,
+        query: Union[str, List[str]],
+        inputs: Optional[Dict[str, Union[StructuredDataset, list]]] = None,
+        **kwargs,
+    ):
+        self._query = query
+        # create an in-memory database that's non-persistent
+        self._con = duckdb.connect(":memory:")
+
+        outputs = {"result": StructuredDataset}
+
+        super(DuckDBQuery, self).__init__(
+            name=name,
+            task_type=self._TASK_TYPE,
+            task_config=None,
+            interface=Interface(inputs=inputs, outputs=outputs),
+            **kwargs,
+        )
+
+    def _execute_query(self, params: list, query: str, counter: int, multiple_params: bool):
+        if any(x in query for x in ("$", "?")):
+            if multiple_params:
+                counter += 1
+                if not counter < len(params):
+                    raise ValueError("Parameter doesn't exist.")
+                if "insert" in query.lower():
+                    # run executemany disregarding the number of entries to store for an insert query
+                    yield QueryOutput(output=self._con.executemany(query, params[counter]), counter=counter)
+                else:
+                    yield QueryOutput(output=self._con.execute(query, params[counter]), counter=counter)
+            else:
+                if params:
+                    yield QueryOutput(output=self._con.execute(query, params), counter=counter)
+                else:
+                    raise ValueError("Parameter not specified.")
+        else:
+            yield QueryOutput(output=self._con.execute(query), counter=counter)
+
+    def execute(self, **kwargs) -> StructuredDataset:
+        """
+        TODO: Enable iterative download after adding the functionality to structured dataset code.
+        """
```

Comment:
can we add a docstring with `:param: name` annotations to this method?

also move the TODO out of the docstring and place it into an inline comment `# TODO`